analyzer/compliance-analyzer/src/main/java/eu/fasten/analyzer/complianceanalyzer/ComplianceAnalyzerPlugin.java

```diff
@@ -0,0 +1,278 @@
+
+    /**
+     * Kubernetes namespace where our demo objects are going to be deployed in.
+     */
+    protected static final String K8S_NAMESPACE = "default";
+
+    /**
+     * How will the patched Job file be called.
+     */
+    private static final String PATCHED_JOB_FILE_NAME = "patchedJob.yaml";
+
+    public ComplianceAnalyzerPlugin(PluginWrapper wrapper) {
+        super(wrapper);
+    }
+
+    @Extension
+    public static class CompliancePluginExtension implements KafkaPlugin, DBConnector {
+        private String consumerTopic = "fasten.RepoCloner.out";
+        private static DSLContext dslContext;
+        private boolean processedRecord = false;
+        private Throwable pluginError = null;
+        private final Logger logger = LoggerFactory.getLogger(CompliancePluginExtension.class.getName());
+        private static RocksDao rocksDao;
+        private String outputPath;
+
+        private String artifactID;
+        private String repoUrl;
+        private String repoPath;
+
+
+        public void setRocksDao(RocksDao rocksDao) {
+            CompliancePluginExtension.rocksDao = rocksDao;
+        }
+
+         @Override
+         public void setDBConnection(DSLContext dslContext) {
+            CompliancePluginExtension.dslContext = dslContext;
+         }
+
+
+        @Override
+        public Optional<List<String>> consumeTopic() {
+            return Optional.of(Collections.singletonList(consumerTopic));
+        }
+
+        @Override
+        public void setTopic(String topicName) {
+            this.consumerTopic = topicName;
+        }
+
+        @Override
+        public void consume(String record) {
+            this.processedRecord = false;
+            this.pluginError = null;
+            var consumedJson = new JSONObject(record);
+            if (consumedJson.has("input")) {
+                consumedJson = consumedJson.getJSONObject("input");
+            }
+            this.repoUrl = consumedJson.getString("repoUrl");
+            var consumedJsonPayload = new JSONObject(record);
+            if (consumedJsonPayload.has("payload")) {
+                consumedJsonPayload = consumedJsonPayload.getJSONObject("payload");
+            }
+            this.repoPath = consumedJsonPayload.getString("repoPath");
+            this.artifactID = consumedJsonPayload.getString("artifactId");
+
+            logger.info("Repo url: " + repoUrl);
+            logger.info("Path to the cloned repo: " + repoPath);
+            logger.info("Artifact id: " + artifactID);
+
+            if (repoUrl == null) {
+                logger.info("No available URL for the specified project.\n");
+                logger.info("Skipping analysis....");
+                return;
+            }
+
+            /**
+             * Connect to Cluster and
+             * start a Kubernetes Job
+             */
+
+            ConnectToCluster();
+            CreateJobConfig();
+            ApplyK8sJob();
+
+            processedRecord = true;
```

Comment:
Fixed in 70778475, I followed the structure of other plugins. Hope this is what you meant @ilyagrishkov.

Reply:
Not exactly. The idea is having a single try-catch with a single catch (Exception e) for the entire consume method. Imagine the case that a message from Kafka topic you have consumed is malformed. It will throw some exception that is not caught by the plugin. It will propagate to the server and execution will crash. Something like that should work:
```java
public void consume(...) {
   try {
   // Everything goes here
   } catch (Exception e) {
      // logger if needed
      setPluginError(e);
   }
}
```

Reply:
Done in f9943e41. Looks good?

Reply:
Yeah, looks good